watertap/unit_models/dewatering.py

```diff
@@ -0,0 +1,177 @@
+    """
+    Dewatering unit block for BSM2
+    """
+
+    CONFIG = SeparatorData.CONFIG()
+    CONFIG.outlet_list = ["underflow", "overflow"]
+    CONFIG.split_basis = SplittingType.componentFlow
+
+    def build(self):
+        """
+        Begin building model.
+        Args:
+            None
+        Returns:
+            None
+        """
+
+        # Call UnitModel.build to setup dynamics
+        super(DewateringUnit, self).build()
+
+        if "underflow" and "overflow" not in self.config.outlet_list:
+            raise ConfigurationError(
+                "{} encountered unrecognised "
+                "outlet_list. This should not "
+                "occur - please use overflow "
+                "and underflow as outlets.".format(self.name)
+            )
+
+        self.p_dewat = Param(
+            initialize=0.28,
+            units=pyunits.dimensionless,
+            mutable=True,
+            doc="Percentage of suspended solids in the underflow",
+        )
+
+        self.TSS_rem = Param(
+            initialize=0.98,
+            units=pyunits.dimensionless,
+            mutable=True,
+            doc="Percentage of suspended solids removed",
+        )
+
+        @self.Expression(self.flowsheet().time, doc="Suspended solid concentration")
+        def TSS(blk, t):
+            return 0.75 * (
+                blk.inlet.conc_mass_comp[0, "X_I"]
+                + blk.inlet.conc_mass_comp[0, "X_P"]
+                + blk.inlet.conc_mass_comp[0, "X_BH"]
+                + blk.inlet.conc_mass_comp[0, "X_BA"]
+                + blk.inlet.conc_mass_comp[0, "X_S"]
+            )
+
+        @self.Expression(self.flowsheet().time, doc="Dewatering factor")
+        def f_dewat(blk, t):
+            return blk.p_dewat * (10 / (blk.TSS[t]))
+
+        @self.Expression(self.flowsheet().time, doc="Remove factor")
+        def f_q_du(blk, t):
+            return blk.TSS_rem / (pyunits.kg / pyunits.m**3) / 100 / blk.f_dewat[t]
+
+        self.non_particulate_components = Set(
+            initialize=[
+                "S_I",
+                "S_S",
+                "S_O",
+                "S_NO",
+                "S_NH",
+                "S_ND",
+                "H2O",
+                "S_ALK",
+            ]
+        )
+
+        self.particulate_components = Set(
+            initialize=["X_I", "X_S", "X_P", "X_BH", "X_BA", "X_ND"]
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.particulate_components,
+            doc="particulate fraction",
+        )
+        def overflow_particulate_fraction(blk, t, i):
+            return blk.split_fraction[t, "overflow", i] == 1 - blk.TSS_rem
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.non_particulate_components,
+            doc="soluble fraction",
+        )
+        def non_particulate_components(blk, t, i):
+            return blk.split_fraction[t, "overflow", i] == 1 - blk.f_q_du[t]
+
```

Comment:
So, for both units, I'm aware that you didn't add `initialize_build` and `calcualte_scaling_factors` methods, likely because the separator handles those. I still wonder though if we should include them here anyway, even if, for example, we just have loggers setup for initialization and run `super().calculate_scaling_factors()`. @andrewlee94 curious to hear your thoughts.

Reply:
If you inherit from another model or base class, then any methods you do not overload will use that of the parent. For initialization this might be OK as long as the additions in the new model are relatively simple. However, for scaling, the new model is adding new variables and constraints (otherwise it wouldn't be a new model), and the parent's scaling methods won't touch these leaving part of the model unscaled.

On a related note, be aware that IDAES plans to rebuild both initialization and scaling this year. The new scaling API is already in place if you want to start looking at it: https://idaes-pse.readthedocs.io/en/latest/reference_guides/initialization/index.html

Reply:
Right, I figured we could technically get away with it since the Separator's initialization and calculate_scaling_factors are essentially being used here, but that it might just be good practice and make for better readability/consistency between models for [new] developers. Furthermore, there are probably some unscaled vars that are added here but simple enough that additional scaling doesn't seem necessary at this point. I think we could put a pin in it for now and revisit if issues arise when analyzing the BSM2 flowsheet later.

And thanks for the note on the new scaling API.